### PR TITLE
Fix Manual Card grid on safari MWPW-117984

### PR DESCRIPTION
**Pull request by meganthecoder**
* Fix "up" grid reverting to default on safari

Resolves: [MWPW-117984](https://jira.corp.adobe.com/browse/MWPW-117984)

**Test URLs:**
- Before: https://main--milo--adobecom.hlx.page/drafts/denli/half-height
- After: https://safari-4up-card--milo--adobecom.hlx.page/drafts/methomas/half-height?martech=off

### CODE REVIEW

**Review by auniverseaway — CHANGES_REQUESTED** [review]
This block name is not ideal. I would opt for, "card"

It is only consider manual when viewed from the context of CaaS... which most people won't be doing.

**Comment by auniverseaway** [comment]
There's also a serious CLS issue going on...

**Comment by auniverseaway** [comment]
Another thing: we have an unspoken (until now) rule that blocks should not affect the section it lives within. The only block that can or should affect a section is section-metadata.

```
 2-up consonant-Wrapper consonant-Wrapper--1200MaxWidth
```
Should not live on the section itself.

This is where you get into competing classes because of the blocks below the section.

**Comment by auniverseaway** [comment]
Also, still not a fan of classes starting with numbers: https://stackoverflow.com/a/449000

**Comment by meganthecoder** [comment]
@auniverseaway Thanks for the feedback. Since these aren't directly about the code for my current ticket, can I address them in a separate ticket?

**Comment by meganthecoder** [comment]
> Also, still not a fan of classes starting with numbers: https://stackoverflow.com/a/449000

When we discussed this, I thought you wanted authors to be about to add "2 up" into the section metadata because it's human-readable. That's why I left it that way. In the code, I take the "2-up" class and turn it into "consonant-CardsGrid--2up" which is necessary for the consonant styles. I can remove the original "2-up" but it seemed unnecessary

**Comment by auniverseaway** [comment]
1. I think it sets a bad precedent to merge something with such poor CLS. The original PR shouldn't have been approved. We fix root problems on Milo. There's no rush to get this in.

2. 2-up on section-metadata is fine (I still prefer alphas, sorry I wasn't clear), we just don't want that other stuff to be firing on the section itself.

**Review by Brandon32 — APPROVED** [review]

File: libs/blocks/card/card.js [hunk 1]
```diff
@@ -17,20 +17,31 @@ const getCardType = (styles) => {
...
 };
 
+const getUpFromSectionMetadata = (section) => {
+  const keyDivs = Array.from(section.querySelectorAll('.section-metadata > div > div:first-child'));
+  const styleDiv = keyDivs.find((keyDiv) => keyDiv.innerText.includes('style'));
+  const styleValues = styleDiv?.nextElementSibling?.textContent.toLowerCase();
+  const styles = styleValues?.split(', ').map((style) => style.replaceAll(' ', '-'));
+
+  return styles?.find((style) => style.includes('-up'));
+};
```
Comment (chrischrischris) [inline libs/blocks/card/card.js:27]: I recently wrote a func to get metadata (aka 2 column table with key val pairs).  It's not checked in yet but you could add it to libs/utils.js:
```suggestion
const getMetadata = (el) => {
  if (!el) return {};

  const md = {};
  el.childNodes.forEach((n) => {
    const key = n.children?.[0]?.textContent?.toLowerCase();
    if (!key) return;

    const val = n.children?.[1]?.textContent?.toLowerCase();
    md[key] = val;
  });
  return md;
};

const getUpFromSectionMetadata = (section) => {
  const sectionMd = getMetadata(section.querySelector('.section-metadata'));
  const styles = sectionMd.style?.split(', ').map((style) => style.replaceAll(' ', '-'));
  return styles?.find((style) => style.includes('-up'));
};
```

**Comment by auniverseaway** [comment]
Do you know why the LCP is so bad, @meganthecoder?

**Comment by auniverseaway** [comment]
CLS is still poor, too.

**Review by chrischrischris — APPROVED** [review]

**Comment by meganthecoder** [comment]
@auniverseaway Here's the analysis for LCP:
Screen Shot 2022-10-06 at 1 24 02 PM
- Reduce unused CSS: We're not using all of the CaaS styles for the Card block because the PMs wanted to reduce the scope for MVP. I believe the intention is to add support for more card types in the future, so this might not be relevant moving forward.
- Avoid chaining requests: This applies to Milo repo, not specifically to this block
- Largest Contentful Pain element: In general there's a decent amount of DOM manipulation going on to create the card. For the image specifically, we're removing the picture element and adding the image url to a div background. I don't see a good way to get around this.

The CLS is in the green. What does it need to be?

**Comment by auniverseaway** [comment]
> The CLS is in the green. What does it need to be?
Ideally 0. There's no reason this block should be causing CLS.

**Comment by meganthecoder** [comment]
Lighthouse for commit 6729513. Moved CaaS styles to deps folder rather than loading from dexter.
safari-4up-card_6729513
Lighthouse for commit e1c21df. Hiding Card block until styles are loaded and CaaS styles in deps folder
safari-4up-card_e1c21df

@auniverseaway I'm not able to get the CLS down to 0 on stage. Moving the styles to deps got the CLS to 0 on my local, but as you can see from the screenshots, that's not the case for the staging environment. I'm not sure what more I can do without completely starting from scratch.

**Comment by meganthecoder** [comment]
Closing in favor of #241

**Pull request closed without merging**